### PR TITLE
docs: Add remaining Implement Tabs

### PR DESCRIPTION
**Pull request by taylorvnoj**
## Motivations

Adding the remaining Implement Tabs on the new docs site

### Added

- any components that have dev centric documentation, gets an Implement tab
- all components that have an Implement tab, now have this comment in the corresponding `component.stories.mdx` file:
```
Sections of this file have been duplicated in the new docs site temporarily.
When updating this file, make sure to also update the corresponding notes file:
packages/site/src/content/[Component]/[Component]Notes.mdx
```

### Changed

- tiny tweaks here and there with grammar, casing, syntax

## Testing

Implement tabs are there for components like `List`, `SideDrawer`. 
The duplication comment in stories aren't showing up on the Design or Docs page

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: docs/components/AnimatedPresence/AnimatedPresence.stories.mdx [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import { Meta } from "@storybook/addon-docs";
 
+{/* Sections of this file have been duplicated in the new docs site temporarily. */}
+{/* When updating this file, make sure to also update the corresponding notes file: */}
+{/* packages/site/src/content/[Component]/[Component]Notes.mdx */}
```
Comment (taylorvnoj) [inline docs/components/AnimatedPresence/AnimatedPresence.stories.mdx:5]: I tried to have this as a multiline comment, so just one `{/* */}` wrapping all the text. However when I did that, the formatting changed on commit to be this (see this commit 74d6798579a1c7064eabc4ffa0caa62a0f44ca84):

```
{/_ Sections of this file have been duplicated in the new docs site temporarily.
When updating this file, make sure to also update the corresponding notes file:
packages/site/src/content/[Component]/[Component]Notes.mdx _/}
```

And that was breaking the build/not actually registering as a comment. I then also tried `<!-- -->` for multi line comment and that didn't work either. The commit's are just empty and don't even show that change.  

So I landed on this `{/* */}` for each line. Seems like single line comments are ok with this format.  This also won't live in the stories files for too long (until docs site is hard launched) so I'm not too concerned. 

If you have another idea let me know. And apologies for the messy commit messages as I was trying to figure this out.

File: docs/components/MultiSelect/MultiSelect.stories.mdx [hunk 2]
```diff
@@ -27,14 +27,15 @@ MultiSelect can be used to help align with common adjacent elements such as
...
-  keys.
-- Users can press the Escape key, click out of the component or click the
-  trigger again to close the menu.
-- Features and behavior are announced to the user when using a screen reader.
+The MultiSelect trigger element contains a downward-pointing-arrow to visually
+hint that it can be expanded.
+
+Features and behavior are announced to the user when using a screen reader.
+
+| Shortcut Keys                        | Action                            |
```
Comment (taylorvnoj) [inline docs/components/MultiSelect/MultiSelect.stories.mdx:35]: Thought it might be a bit more readable to have this table here.

**Review by Aiden-Brine — APPROVED** [review]
Changes look good. As a late joiner to the party we are okay with duplication between the Design tab and the Implement tab? Do we have some guidelines somewhere of what should go where or is that more something that folks have discussed but isn't documented?

**Comment by taylorvnoj** [comment]
@Aiden-Brine great question! We do have a [decision log](https://jobber.atlassian.net/wiki/spaces/TEAMATL/pages/3429990549/Decision+Log#Restructure-dev-focused-Code%2FProps-Tabs) and we did talk separately about this before you joined the project.

> are we okay with duplication between the Design tab and the Implement tab

Right now, yes. Because the Design tab is pulling straight from the Docs pages, if I were to remove the duplication from Docs, that would remove critical info from Storybook which is still in use.  When the hard launch happens, we will remove the duplication.

What should go where has been iterated on, but it's a great callout that decisions have been made without documenting them. I will follow up and add more info to the decision log!

**Pull request merged**